src/gennylib/src/topology.cpp

```diff
@@ -0,0 +1,172 @@
+// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <optional>
+#include <bsoncxx/json.hpp>
+#include <bsoncxx/types.hpp>
+#include <boost/log/trivial.hpp>
+
+#include <gennylib/topology.hpp>
+
+namespace genny {
+
+TopologyVisitor::~TopologyVisitor() {}
+
+TopologyDescription::~TopologyDescription() {}
+
+std::string Topology::nameToUri(const std::string& name) {
+    std::set<std::string> nameSet;
+    // Hostnames returned from some commands have the form configRepl/hostname:port
+    auto strippedName = name.substr(name.find('/') + 1, std::string::npos);
```

Comment:
This is interesting. Some of the names _do_ have the form `hostname:port` without the '/', but this code was still working fine because `npos + 1 == 0` on many implementations, so the substring was "correctly" including the entire string. 

Thanks for pointing this out, I'll update this to be less implementation-dependent.